relay/messages/message.go

```diff
@@ -17,19 +17,23 @@ const (
 	MsgTypeHealthCheck   MsgType = 4
 
 	sizeOfMsgType       = 1
+	sizeOfVersionByte   = 1
 	sizeOfMagicByte     = 4
-	headerSizeTransport = sizeOfMsgType + IDSize                   // 1 byte for msg type, IDSize for peerID
-	headerSizeHello     = sizeOfMsgType + sizeOfMagicByte + IDSize // 1 byte for msg type, 4 byte for magic header, IDSize for peerID
+	headerSizeTransport = sizeOfVersionByte + sizeOfMsgType + IDSize                   // 1 byte for version, 1 byte for msg type, IDSize for peerID
```

Comment:
The version information is enough in the initial message (hello)

Reply:
Correct but we won't be able to swap the whole proto for something else in that case